CONTRIBUTING.md

```diff
@@ -1,48 +1,103 @@
+    * If possible files and flash configuration created with the changes of the Pull request can be used with older firmwares not having the change (downward compatibility). Older firmwares must not break if a newer configuration was stored on the device.
+* If the Pull request changes end user behavoir or introduces new features a new entry in the [CommunityFeatures.md](CommunityFeatures.md) file needs to be created in the preexisting style describing the feature and it's options as a small manual to users. This includes all runtime and compile time flags which shall be named in respective sections.
 
-* Pull requests exclusively fixing insignificant typos in code comments, one-off formatting mistakes, and whitespace will be rejected (though these may be collected together in a bulk fix for a later date).
+## Workflow
 
-* Automated PR submissions by clout/karma-seeking accounts (see also: [Hacktoberfest](https://blog.domenic.me/hacktoberfest/)) will not be tolerated or accepted.
+Please follow the following steps for every pull request to ensure every contribution can be handled as frictionless and transparent as possible: 
 
-## Documentation
+1. Create a [Draft Pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/) including a description on what will be changed and what impact you expect. The title can start with "[Draft] " during development.
+2. Work on the Pull request
+3. Before a Pull request can be considered ready all upstream changes from the ```develop``` branch need to be merged into it. It is the duty of everyone to help make merging into ```develop``` as painless as possible so please try to align if you see that your Pull requests works in a similar area as another one.
+4. Once the Pull request is ready, fulfills all requirements outlined above and is up to date with the ```develop``` branch it can be converted from Draft and marked as ready for review.
+5. If possible every Pull request should have at least two reviews. Reviews from community members not mentioned in the [CODEOWNERS](CODEOWNERS) file should be taken serious and used as an important source of feedback but have no decisional power on what gets merged into the ```develop``` branch.
+5. At least one member of the [CODEOWNERS](CODEOWNERS) file needs to review every pull request while also considering community reviews in their decision.
+    * CODEOWNERS can decline merging a Pull request if it does not fulfill the requirements outlined above. They need to give clear feedback on which requirements have not been met and also provide opportunity to improve the Pull request to meet the requirements within reasonable boundaries (e.g. there are limits on how much work can be expected from a CODEOWNER for a specific Pull request).
+    * If one or more CODEOWNERS are sure the requirements have been met they will merge the change into the ```develop``` branch.
```

Comment:
Honest question: are we making far more work for ourselves by maintaining two branches than we have the capacity to support at this time? For my part I say yes as supporting the one branch is a LOT of work right now even for the two of us officially tasked to do so and a supportive community.

Reply:
Yes, to start with I think we should just continue with `community` as the develop branch. Then we can have "feature freeze" time just before release where we only merge bugfixes, polishes, etc. This flow is used by a lot of FOSS projects with a smaller team.

Reply:
In my head it was just a change of names. `community` currently acts as develop and `community` would be master. Maintenance of `develop` would not change from today and `community` would just be a branch to merge releases into, no other maintenance required.

Alternative suggestion: all development happens on `community` and we just tag releases/release candidates in there?

Reply:
What about making a testing branch, where all PRs can be merged and it's accepted that there might be some bugs? I think that would lower the entry barrier for people who want to test changes by being able to have a single built bin file that incorporates all open PRs. I'm hypothesizing that needing to independently test a build from each PR leads to pretty minimal testing

Largely motivated by my PR that accidentally broke USB hosting earlier this week

Reply:
I suspect breaks are going to be par for the course until we "stabilize". Right now we're in disarray as we refactor and rebuild what's broken out-of-the-box as well as add new features. I know many are expecting to be able to get all the benefits of new features without any of the bugs, but I don't really foresee that happening (or us "releasing") until our work becomes more additive than alterative (if that's a word).

Reply:
Please also see the updated Governance.md, the current wording already reflects the feedback from the first community call. I suggest we try it this way and adapt if it is more work in the end, sounds good?